.github/workflows/build.yml

```diff
@@ -46,21 +46,85 @@ jobs:
           restore-keys: ${{ runner.os }}-m2
 
       # Compile the project
-      - name: Build
+      - name: Compile service
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
         run: mvn -B -V compile
+        working-directory: ./service
 
       # Run the tests
-      - name: Run tests
+      - name: Run service tests
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
         run: mvn -B -V verify
+        working-directory: ./service
 
       # Run Sonar Analysis
-      - name: Analyze with SonarCloud
-        if: ${{ env.SONAR_TOKEN != null && env.SONAR_TOKEN != '' }}
+      - name: Analyze with SonarCloud on service
+        if: ${{ env.SERVICE_SONAR_TOKEN != null && env.SERVICE_SONAR_TOKEN != '' }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SERVICE_SONAR_TOKEN }}
         run: mvn -B -V -Dsonar.coverage.jacoco.xmlReportPaths=target/site/jacoco/jacoco.xml org.jacoco:jacoco-maven-plugin:prepare-agent verify org.jacoco:jacoco-maven-plugin:report org.sonarsource.scanner.maven:sonar-maven-plugin:sonar
+        working-directory: ./service
+  build-ui:
```

Comment:
Could you just have separate build files and maybe there is a way to specify the `working-directory` for the entire file so it does not need to be in every step?

Reply:
This is OBE, I moved it up to the job level